### PR TITLE
feat: add functionality to export/import a dataflow + supporting tests

### LINKED ISSUE #4: Export and import stream setups
All processors should have a standard way to store their internal config so we can have a generic way to export and import either a single step or a whole stream.

### LINKED ISSUE #80: Move all dataformat downloads to bottom of the page

### LINKED ISSUE #82: On steps.php remove table sorting and make order explicit
The order should be execution order based on dependencies. There should be a strict deterministic order even for things which are disconnected sub graphs, or with splits and joins

### PR DESCRIPTION
**Pull request by keevan**
Resolves #4 
Resolves #82 

Resolves #80 because the download options are no longer present.

Note: This does NOT include functionality to import/export individual steps, however the capability to do so is there. Just needs a way to wire it up and call it from the UI.

### CODE REVIEW

File: classes/exportable.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        } else { // Normal http - prevent caching at all cost.
+            header('Cache-Control: private, must-revalidate, pre-check=0, post-check=0, max-age=0');
+            header('Pragma: no-cache');
+        }
+        header('Expires: '. gmdate('D, d M Y H:i:s', 0) .' GMT');
+        header("Content-Type: $this->mimetype\n");
+        header("Content-Disposition: attachment; filename=\"$filename\"");
+    }
+
+
```
Comment (jaypha) [inline classes/exportable.php:75]: Excess whitespace.

File: export.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+$context = context_system::instance();
+
+// Check for caps.
+require_capability('tool/dataflows:managedataflows', $context);
+
+$dataflowid = required_param('dataflowid', PARAM_INT);
+
+$dataflow = new dataflow($dataflowid);
+$dataflow->download();
+
```
Comment (jaypha) [inline export.php:41]: Excess whitespace

File: lang/en/tool_dataflows.php [hunk 3]
```diff
@@ -65,6 +70,8 @@
 $string['new_step'] = 'New Step';
 
 // Warnings / Errors.
+$string['invalidyaml'] = 'The content provided is invalid YAML';
+$string['nodataflowfile'] = 'No dataflow file found';
```
Comment (jaypha) [inline lang/en/tool_dataflows.php:74]: Periods at the end?

**Comment by keevan** [comment]
Just rebased against latest stable branch

**Pull request merged**